functions/ts/Makefile

```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+SHELL=/bin/bash
+TAG := latest
+
+.DEFAULT_GOAL := help
+.PHONY: help
+help: ## Print this help
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: test build npm-ci build-dev push
+npm-ci:
+	npm ci
+
+test: npm-ci ## Run unit tests for all functions
+	npm test
+
+build: test ## Build all function images. Variable 'TAG' is used to specify tag. 'latest' will be used if not set.
+	npm run kpt:docker-build -- --tag $(TAG)
+
+build-dev: test ## Build all function images with tag 'dev'. This is used for local tests.
+	npm run kpt:docker-build -- --tag dev
+
+push: ## Push images to registry. WARN: This operation should only be done in CI environment.
```

Comment:
we should finalize the authz discussion in the doc related to this.

Reply:
I guess this is now only useful for developers that want to push to their own container registry.
But people may argue that developers can directly push the image using `docker push gcr.io/<their-project>/name:tag`. Then this target may not be very useful and we can remove it.

Reply:
It's fine to have the target, but we should prevent pushing to prod registries from a local machine. Only the CD pipeline should have permission.

Reply:
I agree with @mengqiy that there is no need to have a push target.

Reply:
What command does the CD pipeline us to push then? I thought the point here was to abstract away these details.